# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.cpp
# Commit: bff57b2e4b16
# Full Hash: bff57b2e4b16fbcc4b5bf881c66a352ed7e2bd92
# Author: Gabriele Svelto <gsvelto@mozilla.com>
# Date: 2022-09-12 21:40:55
# Regressor Bug: 1789421
# File Overlap Count: 1
# Description:
#   Bug 1789421 - Generate crash reports for content process shutdown hangs only when we can confirm that the process did not shut down in time r=jstutte
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D156838
# ==============================================================================

diff -r f9922dd645d5 -r bff57b2e4b16 dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp	Mon Sep 12 08:56:33 2022 +0000
+++ b/dom/ipc/ContentParent.cpp	Mon Sep 12 08:57:16 2022 +0000
@@ -4289,7 +4289,8 @@
 
     // Generate the report and insert into the queue for submittal.
     if (mCrashReporter->GenerateMinidumpAndPair(this, "browser"_ns)) {
-      mCreatedPairedMinidumps = mCrashReporter->FinalizeCrashReport();
+      mCrashReporter->FinalizeCrashReport();
+      mCreatedPairedMinidumps = true;
     }
   }
 }
@@ -4335,6 +4336,7 @@
   }
 
   if (!KillProcess(otherProcessHandle, base::PROCESS_END_KILLED_BY_USER)) {
+    mCrashReporter->DeleteCrashReport();
     NS_WARNING("failed to kill subprocess!");
   }
 